app/code/Magento/Catalog/Model/View/Asset/Placeholder.php

```diff
@@ -53,24 +54,32 @@ class Placeholder implements LocalInterface
      */
     private $scopeConfig;
 
+    /**
+     * @var Http
+     */
+    private $request;
+
     /**
      * Placeholder constructor.
      *
      * @param ContextInterface $context
      * @param ScopeConfigInterface $scopeConfig
      * @param Repository $assetRepo
+     * @param Http $request
      * @param string $type
      */
     public function __construct(
         ContextInterface $context,
         ScopeConfigInterface $scopeConfig,
         Repository $assetRepo,
+        Http $request,
```

Comment:
The new constructor parameters should be optional in order to avoid backward compatibility issues.
Please check the [Contribution Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#php) (section *Adding a constructor parameter*) for more details.

Reply:
I believe you are referring to the order of the arguments, simply placing it as last would make it optional in this case?

Reply:
Yes, but not exactly.
The idea is to fetch the dependency using `Magento\Framework\App\ObjectManager::getInstance()`, if the new dependency is not provided (i.e. the value of the introduced argument is null).
Please check the example code [here](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter).